packages/syft/src/syft/core/node/common/util.py

```diff
@@ -256,13 +257,20 @@ def upload_to_s3_using_presigned(
             child_kwargs = getattr(data.child, "proxy_public_kwargs", {})
             obj_public_kwargs.update(child_kwargs)
 
+        if isinstance(data, list):
```

Comment:
I think we could go a step further with:

```python
data_shape = (0,)
if hasattr(data, "shape"):
    data_shape = data.shape
elif hasattr(data, "__len__"):
    data_shape = (len(data), )
```

This would ensure that any type with .shape including our own types will work and then as a backup any collection type with a length can be used.

Also I would say its best to preserve the `tuple` of 1 dimension expectation for `.shape` otherwise all the downstream code will get confused if its sometimes an int.